source/solvers/gls_nitsche_navier_stokes.cc

```diff
@@ -610,69 +657,89 @@ GLSNitscheNavierStokesSolver<dim, spacedim>::solve()
 
       if (this->simulation_control->is_output_iteration())
         {
-          std::shared_ptr<Particles::ParticleHandler<spacedim>> solid_ph =
-            solid.get_solid_particle_handler();
-          output_solid_particles(solid_ph);
-          output_solid_triangulation();
+          for (unsigned int i_solid = 0; i_solid < solid.size(); ++i_solid)
+            {
+              std::shared_ptr<Particles::ParticleHandler<spacedim>> solid_ph =
+                solid[i_solid]->get_solid_particle_handler();
+              output_solid_particles();
+              output_solid_triangulation();
```

Comment:
`output_solid_particles` and `output_solid_triangulation` methods both also have a loop over the `i_solid`, so I do not understand the loop here. I think either this loop on `solve` method is unnecessary, or the `i_solid` argument should be passed to the output methods and their loop other `i_solid` should be removed. I would vote for keeping the loop on solve() and passing `i_solid` to the output methods, as they are named `output_solid_xxx` and not `output_solids_xxx` + it would be coherent with the `print_particle_position` method called after for `solid[i_solid]`.

Reply:
I'll keep the loop outside and pass the i_solid to the output methods ^^! Good catch